cli/src/commands/schema/edit.ts

```diff
@@ -115,775 +19,36 @@ export default class EditSchema extends BaseCommand<typeof EditSchema> {
-    const binary = which.sync(env, { nothrow: true }) ? env : info.binary;
-
-    const tmpobj = tmp.fileSync({ prefix: 'schema-', postfix: 'source.json' });
-    // TODO: add a $schema to the document to allow autocomplete in editors such as vscode
-    await writeFile(tmpobj.name, JSON.stringify(branchDetails.schema, null, 2));
-
-    const waitFlag = waitFlags[info.id] || waitFlags[env];
-
-    if (!info.isTerminalEditor && !waitFlag) {
-      this.error(`The editor ${chalk.bold(env)} is a graphical editor that is not supported.`, {
-        suggestions: [
-          `Set the ${chalk.bold('EDITOR')} or ${chalk.bold('VISUAL')} variables to a different editor`,
-          `Open an issue at ${reportBugURL(`Support \`${info.binary}\` editor for schema editing`)}`
-        ]
-      });
-    }
-
-    const args = [waitFlag, tmpobj.name].filter(Boolean);
-    await this.runCommand(binary, args);
-
-    const newSchema = await readFile(tmpobj.name, 'utf8');
-    const result = parseSchemaFile(newSchema);
-    if (!result.success) {
-      this.printZodError(result.error);
-      this.error('The schema is not valid. See the errors above');
-    }
-
-    await this.deploySchema(this.workspace, this.region, this.database, this.branch, result.data);
-
-    // Run pull to retrieve remote migrations
-    await Pull.run([this.branch]);
-  }
-
-  async showInteractiveEditing(branchDetails: Schemas.DBBranch) {
-    this.branchDetails = branchDetails;
-    this.tables = this.branchDetails.schema.tables;
-    await this.showSchema();
-  }
-
-  async showSchema() {
-    this.clear();
-
-    const choices: SelectChoice[] = [
-      this.createSpace() // empty space
-    ];
-    const flatChoices = [...choices];
-
-    const tableChoices: SelectChoice[] = [];
-    const schema: SelectChoice = {
-      name: { type: 'schema' },
-      message: 'Tables',
-      role: 'heading',
-      choices: tableChoices
-    };
-    choices.push(schema);
-    flatChoices.push(schema);
-
-    let index = 0;
-    for (const table of this.tables) {
-      const columnChoices: SelectChoice[] = table.columns.map((column, i) => {
-        if (this.selectItem === column) index = flatChoices.length + i + 1;
-        return {
-          name: { type: 'edit-column', column, table },
-          message: this.getMessageForColumn(table, column)
-        };
-      });
-      columnChoices.push({
-        message: `${chalk.green('+')} Add a column`,
-        name: { type: 'add-column', table },
-        disabled: table.deleted
+    const { workspace, region, database, branch } = await this.getParsedDatabaseURLWithBranch(
+      flags.db,
+      flags.branch,
+      true
+    );
+
+    const details = await getBranchDetailsWithPgRoll(xata, { workspace, region, database, branch });
+
+    if (isBranchPgRollEnabled(details)) {
+      const editNew = new EditSchemaNew(this.argv, config);
+      editNew.launch({
+        workspace,
+        region,
+        database,
+        branch
```

Comment:
We can problably keep editnew in here instead of new file. And only have old one in a separate file.